alacritty_terminal/src/keyboard_motion.rs

```diff
@@ -0,0 +1,767 @@
+        point = expand_semantic(point);
+    }
+
+    point
+}
+
+/// Move by whitespace separated word, like W/B/E/gE in vi.
+fn word<T: EventListener>(
+    term: &mut Term<T>,
+    mut point: Point<usize>,
+    left: bool,
+    start: bool,
+) -> Point<usize> {
+    // Make sure we jump above wide chars
+    point = expand_wide(term, point, left);
+
+    if left == start {
+        // Skip whitespace until right before a word
+        let mut next_point = advance(term, point, left);
+        while !is_boundary(term, point, left) && is_space(term, next_point) {
+            point = next_point;
+            next_point = advance(term, point, left);
+        }
+
+        // Skip non-whitespace until right inside word boundary
+        let mut next_point = advance(term, point, left);
+        while !is_boundary(term, point, left) && !is_space(term, next_point) {
+            point = next_point;
+            next_point = advance(term, point, left);
+        }
+    }
+
+    if left != start {
+        // Skip non-whitespace until just beyond word
+        while !is_boundary(term, point, left) && !is_space(term, point) {
+            point = advance(term, point, left);
+        }
+
+        // Skip whitespace until right inside word boundary
+        while !is_boundary(term, point, left) && is_space(term, point) {
+            point = advance(term, point, left);
+        }
+    }
+
+    point
+}
+
+/// Jump to the end of a wide cell.
+fn expand_wide<T, P>(term: &Term<T>, point: P, left: bool) -> Point<usize>
+where
+    P: Into<Point<usize>>,
+{
+    let mut point = point.into();
+    let cell = term.grid()[point.line][point.col];
+
+    if cell.flags.contains(Flags::WIDE_CHAR) && !left {
+        point.col += 1;
+    } else if cell.flags.contains(Flags::WIDE_CHAR_SPACER)
+        && term.grid()[point.line][point.col - 1].flags.contains(Flags::WIDE_CHAR)
+        && left
+    {
+        point.col -= 1;
+    }
+
+    point
+}
+
+/// Find first non-empty cell in line.
+fn first_occupied_in_line<T>(term: &Term<T>, line: usize) -> Option<Point<usize>> {
+    (0..term.grid().num_cols().0)
+        .map(|col| Point::new(line, Column(col)))
+        .find(|&point| !is_space(term, point))
+}
+
+/// Find last non-empty cell in line.
+fn last_occupied_in_line<T>(term: &Term<T>, line: usize) -> Option<Point<usize>> {
+    (0..term.grid().num_cols().0)
+        .map(|col| Point::new(line, Column(col)))
+        .rfind(|&point| !is_space(term, point))
+}
+
+/// Advance point based on direction.
+fn advance<T>(term: &Term<T>, point: Point<usize>, left: bool) -> Point<usize> {
+    let cols = term.grid().num_cols();
+    if left {
+        point.sub_absolute(cols.0, 1)
+    } else {
+        point.add_absolute(cols.0, 1)
+    }
+}
+
+/// Check if cell at point contains whitespace.
+fn is_space<T>(term: &Term<T>, point: Point<usize>) -> bool {
+    let cell = term.grid()[point.line][point.col];
+    cell.c == ' ' && !cell.flags().contains(Flags::WIDE_CHAR_SPACER)
```

Comment:
We should consider tabs as spaces as well, also I'm not sure how we should move over them.  Test with
` echo -e "\t\thello\n"`, you're also jumping on tabs.

Reply:
We currently allow selecting individual spaces inside tabs and do not treat tabs specially in any other places either. So I suggest for now that we just treat them like whitespace.

However I did fix that tab wasn't treated like whitespace.